### PR TITLE
UCP: Support Dashboard UI under windows

### PR DESCRIPTION
**Pull request by ccw1996**
UCP https://github.com/pingcap/pd/issues/2179
Support Dashboard UI under windows though powershell.

This script is only tested on Windows 10 1909.

### CODE REVIEW

**Comment by nolouch** [comment]
thanks! I will do a test in Windows.

**Comment by HunDunDM** [comment]
You can try to delete `pkg/dashboard/uiserver/empty_assets_handler_windows.go`, and then modify build.ps1 so that it will call your new script.
In addition, please refer to [here](https://github.com/pingcap/community/blob/master/challenge-programs/challenge-program-season-2.md#step-4-submit-the-code) to modify the description of PR, so that you can get the score.

**Comment by ccw1996** [comment]
> You can try to delete `pkg/dashboard/uiserver/empty_assets_handler_windows.go`, and then modify `build.ps1` so that it will call your new script.
> In addition, please refer to [here](https://github.com/pingcap/community/blob/master/challenge-programs/challenge-program-season-2.md#step-4-submit-the-code) to modify the description of PR, so that you can get the score.

Thanks, i will check it and modify

**Comment by ccw1996** [comment]
> You can try to delete `pkg/dashboard/uiserver/empty_assets_handler_windows.go`, and then modify `build.ps1` so that it will call your new script.
> In addition, please refer to [here](https://github.com/pingcap/community/blob/master/challenge-programs/challenge-program-season-2.md#step-4-submit-the-code) to modify the description of PR, so that you can get the score.

i have already modified it , and tested on Windows 10 1909. But i'm still not sure whether normally after delete empty_assets_handler_windows.go. Maybe needs your tested. 

Best Regard

**Comment by HunDunDM** [comment]
I made the following updates:
* Moved `powershell.exe -File embed-dashboard-ui.ps1` before build and fix typo
* Replace the line separator of `embed-dashboard-ui.ps1` with CRLF

File: scripts/embed-dashboard-ui.ps1 [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+if (Test-Path $CACHE_FILE ){
+  echo "  - Cached archive exists: $CACHE_FILE"
+}
+else{
+  echo '  - Cached archive does not exist'
+  echo '  - Download pre-built embedded assets from GitHub release'
+  $DOWNLOAD_URL="https://github.com/pingcap-incubator/tidb-dashboard/releases/download/ui_release_${DASHBOARD_UI_VERSION}/embedded-assets-golang.zip"
+  echo "  - Download ${DOWNLOAD_URL}"
+  curl -UseBasicParsing "${DOWNLOAD_URL}" > embedded-assets-golang.zip
```
Comment (HunDunDM) [inline scripts/embed-dashboard-ui.ps1:33]: The file was not successfully downloaded.

File: scripts/embed-dashboard-ui.ps1 [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+echo '+ Fetch Dashboard Go module'
+go mod download
+
+echo '+ Discover Dashboard UI version'
+
+$DASHBOARD_DIR=$(go list -f "{{.Dir}}" -m github.com/pingcap-incubator/tidb-dashboard)
+echo "  - Dashboard directory: $DASHBOARD_DIR"
+
+$DASHBOARD_UI_VERSION="20200427_1"
```
[the remark this answers is not in the document]
Comment (ccw1996) [reply scripts/embed-dashboard-ui.ps1]: fixed it

File: scripts/embed-dashboard-ui.ps1 (hunk 1 above)
Comment (ccw1996) [reply scripts/embed-dashboard-ui.ps1]: fixed.

**Comment by lhy1024** [comment]
manual test

image
image

**Comment by lhy1024** [comment]
image
manual test

**Comment by HunDunDM** [comment]
I adjusted some output.

**Comment by nolouch** [comment]
thanks @ccw1996

**Pull request merged**